src/uint/boxed/mul.rs

```diff
@@ -12,7 +15,18 @@ impl BoxedUint {
     ///
     /// Returns a widened output with a limb count equal to the sums of the input limb counts.
     pub fn mul(&self, rhs: &Self) -> Self {
-        let mut limbs = vec![Limb::ZERO; self.nlimbs() + rhs.nlimbs()];
+        let size = self.nlimbs() + rhs.nlimbs();
+        let overlap = self.nlimbs().min(rhs.nlimbs());
+
+        if self.nlimbs().min(rhs.nlimbs()) >= 32 {
```

Comment:
Where does 32 come from? Perhaps it could use a constant?

Reply:
It's from benchmarking on this machine, only applying the reduction where it's likely to be faster.